dev/MAINTAINING.md

```diff
@@ -319,3 +319,66 @@ dependencies. May show additional ghc boot libraries.
 ```
 $ nix-shell --run "ghc-pkg dot | dot -Tpdf > streamly.pdf"
 ```
+
+## Regression checking
+
+Please note that you'll need to first go through the `gh auth login` workflow
+and make sure you have the required permissions to the repository.
+
+The following command can be used to run a regression check workflow,
+
+```
+gh workflow run regression-check.yml \
+    -f benchmarks=<benchmarks> \
+    -f fields=<fields> \
+    -f diff-cutoff-percent=<diff-cutoff-percent> \
+    -r <branch>
+```
+
+Examples,
+
+The following command run the workflow for `Data.Array` on branch `array-dev`,
+scrutinizing the field `cuptime`, for any benchmarks that have regressed beyond
+20 percent.
+
+```
+gh workflow run regression-check.yml \
+    -f benchmarks="Data.Array" \
+    -f fields="cputime" \
+    -f diff-cutoff-percent=20 \
+    -r array-dev
+```
+
+The following command run the workflow for `Data.Array and Data.Array.Foreign`
+on branch `array-refactor`, scrutinizing the fields `cuptime` and `bytescopied`,
+for any benchmarks that have regressed beyond 35 percent.
+
+```
+gh workflow run regression-check.yml \
+    -f benchmarks="Data.Array Data.Array.Foreign" \
+    -f fields="cputime bytescopied" \
+    -f diff-cutoff-percent=35 \
+    -r array-refactor
+```
```

Comment:
In the name of `brevity` we can just keep one of these examples, just this one is general enough.